src/main/java/io/r2dbc/postgresql/codec/ZonedDateTimeCodec.java

```diff
@@ -56,14 +56,14 @@ boolean doCanDecode(Format format, PostgresqlObjectId type) {
     ZonedDateTime doDecode(ByteBuf byteBuf, @Nullable Format format, @Nullable Class<? extends ZonedDateTime> type) {
         Objects.requireNonNull(byteBuf, "byteBuf must not be null");
 
-        return ZonedDateTime.parse(ByteBufUtils.decode(byteBuf));
+        return PostgresqlDateTimeFormatter.INSTANCE.parse(ByteBufUtils.decode(byteBuf), ZonedDateTime::from);
     }
 
     @Override
     Parameter doEncode(ZonedDateTime value) {
         Objects.requireNonNull(value, "value must not be null");
 
-        ByteBuf encoded = ByteBufUtils.encode(this.byteBufAllocator, value.toString());
+        ByteBuf encoded = ByteBufUtils.encode(this.byteBufAllocator, value.toOffsetDateTime().toString());
```

Comment:
Trying to understand this change.  Would this be considered a loss in fidelity?

Reply:
Ya, I think so. Take the case where you read it in UTC and then write it back in PST or whatever different timezone. 

IIRC the JDBC driver sets the timezone for the session so we know what it is, and we don't allow the user to change it.